modules/renter/siafile/metadata.go

```diff
@@ -0,0 +1,288 @@
+package siafile
+
+import (
+	"math"
+	"os"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+type (
+	// Metadata is the metadata of a SiaFile and is JSON encoded.
+	Metadata struct {
+		version      [16]byte          // version of the sia file format used
+		fileSize     int64             // total size of the file
+		masterKey    crypto.TwofishKey // masterkey used to encrypt pieces
+		pieceSize    uint64            // size of a single piece of the file
+		trackingPath string            // file to the local copy of the file used for repairing
+		siaPath      string
+
+		// The following fields are the usual unix timestamps of files.
+		modTime    time.Time // time of last content modification
+		changeTime time.Time // time of last metadata modification
+		accessTime time.Time // time of last access
+		createTime time.Time // time of file creation
+
+		// File ownership/permission fields.
+		mode os.FileMode // unix filemode of the sia file - uint32
+		uid  int         // id of the user who owns the file
+		gid  int         // id of the group that owns the file
+
+		// chunkHeaderSize is the size of each of the following chunk's metadata.
+		chunkHeaderSize uint64
+		// chunkBodySize is the size of each of the following chunk's bodies.
+		chunkBodySize uint64
```

Comment:
these two fields and comments aren't perfectly clear.

IF I understand correctly, the chunkHeaderSize is the amount of data in the siafile that is allocated to each chunk, which tells you how to perform random access within the siafile to read and modify chunk metadata.

And then chunkBodySize is the size of the logical data within the chunk so that you know what part of the logical file each chunk index maps to?

Reply:
Hm I'm actually not sure what does fields were about anymore :P I don't think we need both of them. We can always figure out the logical size of a chunk using the erasure coding and the pieceSize.